Corona-Warn-App/src/main/java/de/rki/coronawarnapp/risk/DefaultRiskLevels.kt

```diff
@@ -216,16 +159,313 @@ class DefaultRiskLevels @Inject constructor() : RiskLevels {
+    ): Instant? = exposureHistory
+        .filter { it.riskLevel == riskLevel }
+        .maxOfOrNull { it.dateMillisSinceEpoch }
+        ?.let { Instant.ofEpochMilli(it) }
+
+    private fun aggregateRiskPerDate(
+        dateMillisSinceEpoch: Long,
+        exposureWindowsAndResult: Map<ExposureWindow, RiskResult>
+    ): AggregatedRiskPerDateResult {
+        // 1. Group `Exposure Windows by Date`
+        val exposureWindowsAndResultForDate = exposureWindowsAndResult
+            .filter { it.key.dateMillisSinceEpoch == dateMillisSinceEpoch }
+
+        // 2. Determine `Normalized Time per Date`
+        val normalizedTime = exposureWindowsAndResultForDate.values
+            .sumOf { it.normalizedTime }
+
+        Timber.d("Aggregating result for date $dateMillisSinceEpoch - ${Instant.ofEpochMilli(dateMillisSinceEpoch)}")
+
+        // 3. Determine `Risk Level per Date`
+        val riskLevel = try {
+            appConfig.normalizedTimePerDayToRiskLevelMappingList
+                .filter { it.normalizedTimeRange.inRange(normalizedTime) }
+                .map { it.riskLevel }
+                .first()
+        } catch (e: Exception) {
+            throw ApplicationConfigurationInvalidException(
+                e,
+                "Invalid config for normalizedTimePerDayToRiskLevelMapping"
+            )
+        }
+
+        Timber.d("riskLevel: ${riskLevel.name} (${riskLevel.ordinal})")
+
+        // 4. Determine `Minimum Distinct Encounters With Low Risk per Date`
+        val minimumDistinctEncountersWithLowRisk = minimumDistinctEncountersForRisk(
+            exposureWindowsAndResultForDate,
+            RiskCalculationParametersOuterClass.NormalizedTimeToRiskLevelMapping.RiskLevel.LOW
+        )
+
+        Timber.d("minimumDistinctEncountersWithLowRisk: $minimumDistinctEncountersWithLowRisk")
+
+        // 5. Determine `Minimum Distinct Encounters With High Risk per Date`
+        val minimumDistinctEncountersWithHighRisk = minimumDistinctEncountersForRisk(
+            exposureWindowsAndResultForDate,
+            RiskCalculationParametersOuterClass.NormalizedTimeToRiskLevelMapping.RiskLevel.HIGH
+        )
+
+        Timber.d("minimumDistinctEncountersWithHighRisk: $minimumDistinctEncountersWithHighRisk")
+
+        return AggregatedRiskPerDateResult(
+            dateMillisSinceEpoch,
+            riskLevel,
+            minimumDistinctEncountersWithLowRisk,
+            minimumDistinctEncountersWithHighRisk
+        )
+    }
+
+    private fun minimumDistinctEncountersForRisk(
+        exposureWindowsAndResultForDate: Map<ExposureWindow, RiskResult>,
+        riskLevel: RiskCalculationParametersOuterClass.NormalizedTimeToRiskLevelMapping.RiskLevel
+    ): Int =
+        exposureWindowsAndResultForDate
+            .filter { it.value.riskLevel == riskLevel }
+            .map { "${it.value.transmissionRiskLevel}_${it.key.calibrationConfidence}" }
```

Comment:
Can you explain the idea behind this?

Reply:
_4. Determine Minimum Distinct Encounters With Low Risk per Date : for all Exposure Windows of the same Date , count the
distinct combinations of Transmission Risk Level , and Callibration Confidence of all Exposure Windows with Risk
Level calculated as Low_ (max docu)